app/src/main/java/org/mozilla/reference/browser/components/Toolbar.kt

```diff
@@ -0,0 +1,90 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.reference.browser.components
+
+import android.content.Context
+import android.content.Intent
+import mozilla.components.browser.domains.autocomplete.ShippedDomainsProvider
+import mozilla.components.browser.menu.BrowserMenuBuilder
+import mozilla.components.browser.menu.item.BrowserMenuItemToolbar
+import mozilla.components.browser.menu.item.SimpleBrowserMenuCheckbox
+import mozilla.components.browser.menu.item.SimpleBrowserMenuItem
+import mozilla.components.browser.session.SessionManager
+import mozilla.components.feature.session.SessionUseCases
+import org.mozilla.reference.browser.R
+import org.mozilla.reference.browser.ext.share
+import org.mozilla.reference.browser.settings.SettingsActivity
+
+/**
+ * Component group for all functionality related to the browser toolbar.
+ */
+class Toolbar(
+    private val context: Context,
+    private val sessionUseCases: SessionUseCases,
+    private val sessionManager: SessionManager
+) {
+
+    /**
+     * Helper class for building browser menus.
+     */
+    val menuBuilder by lazy { BrowserMenuBuilder(menuItems) }
+
+    /**
+     * Provides autocomplete functionality for shipped / provided domain lists.
+     */
+    val shippedDomainsProvider by lazy {
```

Comment:
nit: In theory we could use this elsehwere (where we might want to autocomplete URLs... although it's not clear to me what that is in the reference-browser, other than the url bar), not just within the toolbar, right? If the answer is 'yes', maybe a utils group is a better home?

Reply:
Agreed, we'll end up moving things around quite a bit as we go along. I'd suggest doing this later when we need to.